daffodil-runtime1/src/main/scala/org/apache/daffodil/infoset/SAXInfosetInputter.scala

```diff
@@ -0,0 +1,192 @@
+
+import org.apache.daffodil.api.DFDL
+import org.apache.daffodil.api.DFDL.DaffodilUnhandledSAXException
+import org.apache.daffodil.api.DFDL.DaffodilUnparseErrorSAXException
+import org.apache.daffodil.dpath.NodeInfo
+import org.apache.daffodil.infoset.InfosetInputterEventType.EndDocument
+import org.apache.daffodil.util.Maybe.One
+import org.apache.daffodil.util.MaybeBoolean
+import org.apache.daffodil.util.Misc
+import org.apache.daffodil.xml.XMLUtils
+
+/**
+ * The SAXInfosetInputter consumes infosetEvent objects from the DaffodilInputContentHandler class
+ * and converts it to an event that the Dataprocessor unparse can use. This class contains two
+ * infosetEvent objects that contain the current event the unparse method is processing, and the
+ * next event to be processed after.
+ *
+ * This class together with the DaffodilInputContentHandler use coroutines to ensure that only one event,
+ * at a time, is passed between the two classes. The following is the general process:
+ *
+ * - the run method is called, with a StartDocument event already loaded on the inputter's queue.
+ * This is collected and stored in the nextEvent method, and the inputter's next method is called
+ * to populate the currentEvent and load the nextEvent
+ * - The dp.unparse method is called, and it calls hasNext to make sure an event exists to be
+ * processed and then queries the currentEvent, after it is done with the currentEvent, it calls
+ * inputter.next to get the next event, and that copies the nextEvent into the currentEvent and
+ * transfser control to the contentHandler to load the nextEvent
+ * - That continues until the currentEvent contains an EndDocument event, at which point, the
+ * nextEvent is clear, endDocumentReceived is set to true and hasNext is set to false
+ * - This ends the unparse process, and the unparseResult and/or any Errors are set on the event,
+ * and we call resumeFinal passing along that element, terminating this thread and resuming the
+ * contentHandler for the last time.
+ *
+ * @param inputContentHandler producer coroutine that sends infosetEvent to this class
+ * @param dp dataprocessor that we use to kickstart the unparse process and that consumes the
+ *           currentEvent
+ * @param output  outputChannel of choice where the unparsed data is stored
+ */
+class SAXInfosetInputter(
+  inputContentHandler: DFDL.DaffodilInputContentHandler,
+  dp: DFDL.DataProcessor,
+  output: DFDL.Output)
+  extends InfosetInputter with DFDL.ConsumerCoroutine {
+  // allows support for converting relative URIs in data to absolute URIs, this is mainly use
+  // for TDML as the tests allow relative URIs. This can be set to true by calling the
+  // inputContentHandler.enableInputterUriAbsolutization()
+  var enableUriAbsolutization: Boolean = false
+
+  var endDocumentReceived = false
+  val currentEvent: DFDL.SaxInfosetEvent = new DFDL.SaxInfosetEvent
+  val nextEvent: DFDL.SaxInfosetEvent = new DFDL.SaxInfosetEvent
+
```

Comment:
Can these vals be made private? Should help IDE autocompletion if we get better at marking things private that can be private. I think scala compiler can also make some optimizations if it knows that nothign outside this class can access a variable.